tests/test_browser.py

```diff
@@ -4652,10 +4652,25 @@ def test_base64_atob_fallback(self):
 
   # Tests that SINGLE_FILE works as intended in generated HTML (with and without Worker)
   def test_single_file_html(self):
-    self.btest('emscripten_main_loop_setimmediate.cpp', '1', args=['-s', 'SINGLE_FILE=1', '-s', 'WASM=1'], also_proxied=True)
+    self.btest('single_file_static_initializer.cpp', '1', args=['-s', 'SINGLE_FILE=1', '-s', 'WASM=1'], also_proxied=True)
```

Comment:
why is the source file changed here?

Reply:
The file `emscripten_main_loop_setimmediate.cpp` does not have any use of static data, so if there is a failure in memory initializer loading (like was the case of this bug), then testing against that file won't catch it. That is why it was switched to testing code that does stress memory initialization.

Then, there is an existing test

```
  @requires_threads
  def test_emscripten_main_loop_setimmediate(self):
    for args in [[], ['--proxy-to-worker'], ['-s', 'USE_PTHREADS=1', '-s', 'PROXY_TO_PTHREAD=1']]:
      self.btest('emscripten_main_loop_setimmediate.cpp', '1', args=args)
```
That already does test `emscripten_main_loop_setimmediate()` API both in main thread and in worker, so having extra testing for that API in SINGLE_FILE mode probably would not be that important.